thirdeye/thirdeye-frontend/app/pods/manage/alert/explore/route.js

```diff
@@ -413,15 +374,20 @@ export default Route.extend({
             anomalyData,
             baselineOptions: [baselineOptions[0], ...newWowList]
           });
-          // If alert has dimensions set, load them into graph
-          if (exploreDimensions) {
-            return fetch(topDimensionsUrl).then(checkStatus).then((allDimensions) => {
-              const newtopDimensions = getTopDimensions(subD, allDimensions, alertDimension);
-              controller.setProperties({
-                topDimensions: newtopDimensions,
-                isMetricDataLoading: false
-              });
-            });
+          return fetch(metricDataUrl).then(checkStatus);
+        })
+        // Fetch and load graph metric data. Stop spinner/loader
+        .then((metricData) => {
```

Comment:
Looks like we have multiple nested `.then`s here. 
Let's start looking at async/await documentation for a future refactor: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function

#### Example implementations in our code:
- <strong>Component: </strong> https://github.com/linkedin/pinot/blob/master/thirdeye/thirdeye-frontend/app/pods/components/te-modal/component.js#L66
- <strong>Test: </strong> https://github.com/linkedin/pinot/blob/master/thirdeye/thirdeye-frontend/tests/acceptance/rootcause-test.js#L49

Reply:
I would say they are multiple "chained" `.then`s. Wondering if we should bundle some of them in RSVP hash. I'll leave a recommendation note.